config.py

```diff
@@ -20,6 +21,17 @@ class ConfigTemplater(object):
     def add_template(self, template):
         self.t[template.name] = template
 
+    def additional_default_options(self):
+        output = ''
+        base_options = ['redispatch', 'http-server-close', 'dontlognull']
+        if 'HAPROXY_DEFAULT_OPTIONS' in os.environ:
+            opts = set(re.split('\s*,\s',
+                                os.getenv('HAPROXY_DEFAULT_OPTIONS')))
+            for opt in [x for x in opts if x not in base_options]:
```

Comment:
How about we remove the default options from the base template, and then set `redispatch,http-server-close,dontlognull` as the default option set. This way, if someone wants to remove some of the options, they can.